core/src/main/java/org/opensearch/remote/metadata/client/DataObjectResponse.java

```diff
@@ -90,6 +90,16 @@ public RestStatus status() {
         return this.status;
     }
 
+    /**
+     * The {@link XContentParser} extends {@link java.io.Closeable} so it'll be closed after it's been read.
+     * In {@link GetDataObjectResponse#getResponse()} it's been read, after it's read the parser object
+     * is not null but the content in the parser is set to null. If it's read again e.g. clients who
+     * use parser to deserialize to an object, then a deserialization exception will be thrown.
+     */
+    protected void setParserToNull() {
+        this.parser = null;
+    }
```

Comment:
I don't think a method "set to null" is appropriate at all.

Why is this `protected`?  In what situation would any subclass ever use this?

Much time was spent evaluating the possible code paths for null vs. non-null based on the `final` modifier and this completely changes that entire set of assumptions.

Passing around `null` is generally bad and it was carefully evaluated and tested previously.  This change seems an unnecessary shortcut solution.

Please convince me this is needed, otherwise I won't approve this.